openaz-xacml/src/main/java/org/apache/openaz/xacml/std/jaxp/JaxpResponse.java

```diff
@@ -102,18 +103,16 @@ public static JaxpResponse load(File fileXmlResponse) throws ParserConfiguration
             return null;
         }
 
-        NodeList nodeListRoot = document.getChildNodes();
-        if (nodeListRoot == null || nodeListRoot.getLength() == 0) {
-            logger.warn("No child elements of the XML document");
-            return null;
-        }
-        Node nodeRoot = nodeListRoot.item(0);
-        if (nodeRoot == null || nodeRoot.getNodeType() != Node.ELEMENT_NODE) {
-            logger.warn("Root of the document is not an ELEMENT");
-            return null;
-        }
+        Node nodeRoot = document.getDocumentElement();
 
-        JAXBContext context = JAXBContext.newInstance(ResponseType.class);
+        if(context == null) {
```

Comment:
Where is your own unmarshal code ? it is being performed here inside the response/request class. I totally see scalability issue in this patch and not comfortable in giving thumbs up, What you need is single instance of context (which is a good thought) but in code you are doing it at two different places pretty much similar way just flipping class type. If you really need to parameterize class type, again create singleton class and initialize it with custom class type (request or response). But, don't patch or modify existing class with synchronized block. 

IMHO, we should avoid synchronization, whenever possible and have things thread local and immutable.